### PR TITLE
Fix concurrent map write issue at AggregateWatcher stop

### PR DESCRIPTION
**Pull request by Sindica**

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/watch/watch.go [hunk 1]
```diff
@@ -195,11 +195,16 @@ func (a *AggregatedWatcher) Stop() {
 	if !a.stopped {
 		a.stopped = true
 		a.allowWatcherReset = false
+
+		a.mapLock.RLock()
```
Comment (yb01) [inline staging/src/k8s.io/apimachinery/pkg/watch/watch.go:199]: not sure why to write to the stop channel needs unlock the watcher map, some comment can help here.

**Review by yb01 — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/apimachinery/pkg/watch/watch.go (hunk 1 above)
Comment (Sindica) [reply staging/src/k8s.io/apimachinery/pkg/watch/watch.go:199]: Updated due to CR feedback.

**Pull request merged**